packages/confirmation-modal/src/ConfirmationModal.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@leafygreen-ui/emotion';
+import Button, { Variant } from '@leafygreen-ui/button';
+import Modal from '@leafygreen-ui/modal';
+import { uiColors } from '@leafygreen-ui/palette';
+import TextInput from '@leafygreen-ui/text-input';
+
+const titleStyle = css`
+  color: ${uiColors.gray.dark2};
+  font-size: 24px;
+  font-weight: bold;
+  line-height: 25px;
+
+  margin-bottom: 10px;
+`;
+
+const baseModalStyle = css`
+  width: 600px;
+  padding: initial;
+`;
+
+const contentStyle = css`
+  color: ${uiColors.gray.dark1};
+  font-family: Akzidenz, ‘Helvetica Neue’, Helvetica, Arial, sans-serif;
+  font-size: 16px;
+  letter-spacing: 0;
+  line-height: 24px;
+
+  padding: 36px;
+`;
+
+const textEntryInputStyle = css`
+  width: 300px;
+  padding-top: 14px;
+`;
+
+const footerStyle = css`
+  &:before {
+    content: '';
+    display: block;
+    position: relative;
+    width: 100%;
+    border-top: 1px solid ${uiColors.gray.light2};
+  }
+`;
+
+const footerContentStyle = css`
+  display: flex;
+  justify-content: right;
+  flex-direction: row-reverse;
+  column-gap: 5px;
+  padding: 16px 24px;
+`;
+
+interface ConfirmationModalProps {
+  title: string;
+  children: React.ReactNode;
+  open?: boolean;
+  setOpen?: (open: boolean) => void;
+  className?: string;
+  primaryActionProps: ActionProps;
+  secondaryActionProps: ActionProps;
```

Comment:
I might be missing something, so let me know if I am.

I believe that the "secondary action" is always going to be "Cancel" in confirmation modals, and always close the modal, so I don't think it should be configurable. I might suggest that the cancel button fire an `onClose` callback for additional handling an application might need for clearing a form, or updating state. That would probably need to be handled within the `Modal` component.

Reply:
I decided to merge `setOpen` and `onClose` since they're kind of redundant + the modal never calls `setOpen(true)`.

Reply:
Would be nice if the base Modal component were to merge `setOpen` and `shouldClose`, but that's a discussion for another time.